### ISSUE TITLE
[ Product Extensibility] Use SelectTree to create a component to select a product brand

### ISSUE THREAD

**Issue by nathanss** [woothemes/woocommerce#39205]
This is part of of issue [#248](https://github.com/woocommerce/woocommerce-brands/issues/248)

Depends on #39204.

**Description**

We should replace the brands placeholder block with the actual field. We can use the [SelectTree](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/experimental-select-tree-control/select-tree.tsx) component with the `multiple` prop set to `false`.

In our previous attempt to migrate the plugin, [[we also had to do some adjustments to the Rest API](https://github.com/nathanss/vienna-brands/blob/main/vienna-brands.php#L71-L102)](https://github.com/nathanss/vienna-brands/blob/main/vienna-brands.php#L71-L102), because there were some inconsistencies with the way it worked and how we were persisting the object. Since we are now on the brands plugin codebase, we can tweak the API to make it easier to work on both sides.

It's possible to fetch all the product brands on the front-end by calling 
```javascript
wp.data.select('core').getEntityRecords('taxonomy', 'product_brand')
```

PS: Creating a new brand directly from the field will be handled in another issue.

**Acceptance criteria**

- [ ] When opening the product editor with the brands plugin enabled, I can use the SelectTree component to select a product brand from an already existing list (creating new brands from the inside the editor will be implemented in future) and it's persisted successfully
